src/devices/GamepadXRInputSource.js

```diff
@@ -31,6 +31,24 @@ class XRRemappedGamepad {
       map = {};
     }
 
+    // Apply user-agent-specific overrides to the mapping when applicable.
+    if (map.userAgentOverrides) {
+      for (let agent in map.userAgentOverrides) {
+        if (navigator.userAgent.includes(agent)) {
+          let override = map.userAgentOverrides[agent];
+
+          for (let key in override) {
```

Comment:
This was the slightest bit confusing, so a comment might be helpful.  For a moment I was wondering how you would ever have an override when there was no key already in the map, but then I realized that you're treating it as an implied key if the array ordering happened to already match the WebXR requirements.